managed/resource_cluster.go

```diff
@@ -1853,6 +1965,19 @@ func (r resourceCluster) Update(ctx context.Context, req tfsdk.UpdateResourceReq
 		}
 	}
 
+	connectionPoolingChanged := false
+
+	// Disable Connection Pooling if the desired state is set to 'Disabled and it is enabeld currently
+	if strings.EqualFold(plan.DesiredConnectionPoolingState.Value, "Enabled") && (plan.DesiredConnectionPoolingState.Unknown || strings.EqualFold(plan.DesiredConnectionPoolingState.Value, "Disabled")) {
+		// Disable Connection Pooling
```

Comment:
You need to set the state to disabled irrespective of the current state. Also, if desired state is no unknown you are checking `desired state == Enabled && desired state == Disabled`. This is never going to happen. I think you can just set it to disabled if it is unknown or disabled.